agent/agent_test.go

```diff
@@ -1819,7 +1825,16 @@ tasks:
 		Description:            "task failed",
 		AddFailureMetadataTags: []string{"failure_tag0", "failure_tag1", "failure_tag2"},
 	}
+
+	s.Nil(s.tc.userEndTaskResp)
 	s.tc.setUserEndTaskResponse(resp)
+	s.NotNil(s.tc.userEndTaskRespOriginatingCommand)
+	s.Equal(s.tc.userEndTaskRespOriginatingCommand.FullDisplayName(), "initial task setup")
```

Comment:
nit: the order of parameters to `s.Equal` is the expected value (`"initial task setup"`) first, so it would be:
```go
	s.Equal("initial task setup", s.tc.userEndTaskRespOriginatingCommand.FullDisplayName())
```